python_modules/libraries/dagster-airflow/dagster_airflow/dagster_job_factory.py

```diff
@@ -249,3 +253,106 @@ def make_airflow_example_dags():
+        }
+        asset_key_to_task_ids = {
+            dag.dag_id: leaf_nodes
+        }
+
+    output_mappings=set()
+    keys_by_output_name = {}
+    internal_asset_deps={}
+
+    visit_dict = {}
+    upstream_deps = set()
+
+    def find_upstream_dependency(solid_name: str) -> None:
+        # DFS to find all upsteam dependencies
+        # this node has been visited
+        if visit_dict[solid_name]:
+            return
+        # mark node as visted
+        visit_dict[solid_name] = True
+        # traverse upstream nodes
+        for output_handle in graph.dependency_structure.all_upstream_outputs_from_node(solid_name):
+            forward_node = output_handle.node_name
+            match = False
+            # find any assets produced by upstream nodes and add them to the internal asset deps
+            for asset_key in asset_key_to_task_ids:
+                if forward_node.replace('airflow_', "") in asset_key_to_task_ids[asset_key]:
+                    upstream_deps.add(AssetKey(asset_key))
+                    match = True
+            # don't traverse past nodes that have assets
+            if not match:
+                find_upstream_dependency(forward_node)
+
+    for asset_key in asset_key_to_task_ids:
+        asset_upstream_deps = set()
+        for task_id in asset_key_to_task_ids[asset_key]:
+            visit_dict = {s.name: False for s in graph.solids}
+            upstream_deps = set()
+            find_upstream_dependency(f"airflow_{task_id}")
+            for dep in upstream_deps:
+                asset_upstream_deps.add(dep)
+            internal_asset_deps[f"result_airflow_{task_id}"] = upstream_deps
+            keys_by_output_name[f"result_airflow_{task_id}"] = AssetKey(asset_key)
+            output_mappings.add(OutputMapping(
+                graph_output_name=f"result_airflow_{task_id}",
+                mapped_node_name=f"airflow_{task_id}",
+                mapped_node_output_name="airflow_task_complete" # Default output name
+            ))
+
+        # the tasks for a given asset should have the same internal deps
+        for task_id in asset_key_to_task_ids[asset_key]:
+            internal_asset_deps[f"result_airflow_{task_id}"] = asset_upstream_deps
+
+    new_graph = GraphDefinition(
+        name=graph.name,
+        node_defs=graph.node_defs,
+        dependencies=graph.dependencies,
+        output_mappings=list(output_mappings),
+    )
+
+    asset_def = AssetsDefinition.from_graph(
+        graph_def=new_graph,
+        partitions_def=TimeWindowPartitionsDefinition(
+            cron_schedule=cron_schedule,
+            timezone=dag.timezone.name,
+            start=start_date.strftime("%Y-%m-%d"),
+            fmt="%Y-%m-%d"
+        ) if isinstance(cron_schedule, str) and is_valid_cron_schedule(cron_schedule) else None,
+        resource_defs=job.resource_defs,
+        group_name=dag.dag_id,
+        keys_by_output_name=keys_by_output_name,
+        internal_asset_deps=internal_asset_deps,
```

Comment:
Let me know if I'm misunderstanding, but I don't think we need to provide this argument. This argument is mainly for when there's structure within _a single op_.  If it's within a graph, then I believe we automatically infer the asset dependencies from the op dependencies within the graph.

Reply:
@OwenKephart can confirm or deny what I'm saying

Reply:
Without this arg the SDA graph will look like:
![Screen Shot 2023-01-27 at 2 37 12 PM](https://user-images.githubusercontent.com/5943242/215181636-3e8dc544-c28f-4c86-92d2-b99c38951785.png)
But maybe there's another spot to declare the dependencies?

Reply:
Ah OK - sounds like it is required. I think we should modify `AssetsDefinition.from_graph` to do this on its own, but not a blocker for this PR.

Reply:
I don't think it's intended that we'd have this behavior -- we do have [code in place](https://sourcegraph.com/github.com/dagster-io/dagster/-/blob/python_modules/dagster/dagster/_core/definitions/asset_layer.py?L386:5) to parse out the dependencies within graph-backed assets, which apparently I wrote but have no recollection of.

so this is probably a bug but a good workaround in the meantime is indeed just doing what you're doing